dashboard/app/models/user.rb

```diff
@@ -362,6 +364,17 @@ def save_email_reg_partner_preference
     end
   end
 
+  # Puts teachers directly into the progress table v2 view when new account is created.
+  def save_show_progress_table_v2
+    if email.present?
+      user = User.find_by_email_or_hashed_email(email)
+      if user&.teacher?
+        user.show_progress_table_v2 = true
+        user.save!
+      end
+    end
```

Comment:
Would this work instead? I don't love looking the user up all over again from an instance method.
```suggestion
    if teacher?
      show_progress_table_v2 = true
      save!
    end
```

Reply:
Good point!  This new code mostly works, I just had to modify it a bit.   We will see if it passes drone!  I did check the ruby tests that were failing before and it seemed to pass those.